File: src/main/java/net/orcinus/galosphere/network/PlayCooldownSoundPacket.java
package net.orcinus.galosphere.network;

import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;
import net.minecraft.network.protocol.common.custom.CustomPacketPayload;

public record PlayCooldownSoundPacket() implements CustomPacketPayload {
    public static final StreamCodec<FriendlyByteBuf, PlayCooldownSoundPacket> CODEC = CustomPacketPayload.codec(PlayCooldownSoundPacket::write, PlayCooldownSoundPacket::new);
    public static final Type<PlayCooldownSoundPacket> TYPE = CustomPacketPayload.createType("play_cooldown_sound");

    public PlayCooldownSoundPacket(FriendlyByteBuf buf) {
        this();
    }

    public void write(FriendlyByteBuf buf) {
    }

    @Override
    public Type<? extends CustomPacketPayload> type() {
        return TYPE;
    }
}

File: src/main/java/net/orcinus/galosphere/network/PlayCooldownSoundPacket.kt
package net.orcinus.galosphere.network

import net.minecraft.network.FriendlyByteBuf
import net.minecraft.network.protocol.common.custom.CustomPacketPayload
import net.orcinus.galosphere.Galosphere

class PlayCooldownSoundPacket() : CustomPacketPayload
{
	constructor(buf: FriendlyByteBuf) : this()

	fun write(buf: FriendlyByteBuf)
	{
	}

	override fun type () = TYPE

	companion object
	{
		@JvmField
		val CODEC = CustomPacketPayload.codec(PlayCooldownSoundPacket::write, ::PlayCooldownSoundPacket)
		@JvmField
		val TYPE = CustomPacketPayload.Type<PlayCooldownSoundPacket>(Galosphere.id("play_cooldown_sound"))
	}
}